### DISCUSSION TITLE
Better error messaging when passthrough copy writes to the same output as a template (or vice versa)

### DISCUSSION THREAD

**Discussion by alesandroortiz** [11ty/eleventy#4084] [Enhancement Queue]
**Describe the bug**
Transform functions receive files within passthrough copy directories even though the transform output is ultimately ignored.

Updating transforms to only receive relevant files will likely lead to performance improvements and prevent confusion when creating filters within transforms. (e.g. I thought I needed to filter out passthrough copy dirs; turns out that is unnecessary even if the transform receives them.)

Although unlikely to be used as such by a developer, if a transform performs actions other than returning the transformed content, then files in the directory may be accidentally modified. (See https://github.com/11ty/eleventy/issues/344#issuecomment-485019492 where this behavior is used to intentionally modify files within a passthrough dir; similarly, other users may do so accidentally.)

**To Reproduce**
1. Create a directory with a file inside. (e.g. `mkdir keep-intact && echo "Important File" > keep-intact/important-file.html"`
2. Set the directory to a passthrough directory via `addPassthroughCopy("keep-intact")`.
3. Add a transform which replaces all files with arbitrary output (e.g. `return "Transformed file.";`).
4. Run `npx eleventy build` to build the site.

For easy repro, I've created a PoC project:
1. Clone PoC repo to your environment: https://github.com/thephpjedi/eleventy-bug-1
2. Run `npx eleventy build` to build the site.

**Expected behavior**
Transforms *do not* receive any files within passthrough copy directories, since the transform will not occur.

In output from PoC project build, note that `Transform received` *does not* include `keep-intact` passthrough directory.
```
thephpjedi@Alurion:~/eleventy-poc$ npx eleventy build
Transform received: _site/hello-world/index.html
Writing _site/hello-world/index.html from ./hello-world.html.
Writing _site/keep-intact/important-file/index.html from ./keep-intact/important-file.html.
Copied 1 item and Processed 2 files in 0.11 seconds (v0.9.0)
```

**Observed behavior**
Transforms receive all files within passthrough copy directories. The transform return value is ignored, so the file is not transformed, but the transform is still called unnecessarily.

In output from PoC project build, note that `Transform received` includes `keep-intact` passthrough directory.
```
thephpjedi@Alurion:~/eleventy-poc$ npx eleventy build
Transform received: _site/hello-world/index.html
Transform received: _site/keep-intact/important-file/index.html
Writing _site/hello-world/index.html from ./hello-world.html.
Writing _site/keep-intact/important-file/index.html from ./keep-intact/important-file.html.
Copied 1 item and Processed 2 files in 0.11 seconds (v0.9.0)
```

**Environment:**
 - OS and Version: Ubuntu 18.04.2 LTS on Windows Subsystem for Linux (`Linux Alurion 4.4.0-17763-Microsoft #379-Microsoft Wed Mar 06 19:16:00 PST 2019 x86_64 x86_64 x86_64 GNU/Linux`)
 - Eleventy Version 0.9.0

**Comment by zachleat** [comment]
Howdy! I am slowly making my way through old issues—I see this one still seems to do what you’re reporting but I’m not sure it’s wrong?

I see 3 files in the output:

* `_site/hello-world/index.html` (transformed)
* `_site/keep-intact/important-file/index.html` (transformed)
* `_site/keep-intact/important-file.html` (copied, not transformed)

I don’t see anything wrong there. **However** if I rename `keep-intact/important-file.html` to `keep-intact/index.html` then I do see a problem, two files in the output:

* `_site/hello-world/index.html` (transformed)
* `_site/keep-intact/index.html` (copied, not transformed)

But likely there was a race condition here, where the copy finished after the template. I wouldn’t be surprised if that was inconsistent as a project grows.

I think I would expect an error message if a passthrough copy and a template try to write to the same place. Related to #1677

**Comment by zachleat** [comment]
> This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

> View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!
